txn_writer.go

```diff
@@ -473,3 +473,36 @@ func hasDotGit(path string) bool {
 	_, err := os.Stat(gitfilepath)
 	return err == nil
 }
+
+// AddToManifest adds new dependencies to the existing manifest.
+func (sw *SafeWriter) AddToManifest(root string, a *Manifest) error {
```

Comment:
so the big goal of the safe writer is to do grouped writes, rolling them all back together if there's a problem with any of them. the big FIXME here is about bringing us closer to that, but this just sorta codifies what's already wrong about the code that's there by creating a separate method.

instead, let's consider an approach that keeps everything rolled in together - say, adding another parameter to `NewSafeWriter()` specifically for the appender manifest, then smoothly integrating what's in this method into the overall flow of `SafeWriter.Write()`.

does that make sense?

Reply:
Thanks for the review @sdboyer 
I think I got it. I will make the changes and update the PR.